python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+    all_related_prs = get_prs(repository, earliest_pr_date, main_branch)
+    # get commits for selected prs with backport label
+    prs_with_backport = get_prs_by_label(all_related_prs, "backport")
+    print(f"### OK {len(prs_with_backport)} PRs with backport label found. PR list is as below### ")
+    for pr in prs_with_backport:
+        print(f"No:{pr.number} Title:{pr.title}\n")
+    # cherrypick all commits with backport label
+    print(f"Cherry-picking PRs on {release_branch_name}...")
+    cherry_pick_prs(prs_with_backport)
+    print(f"OK Cherry pick completed for all PRs on branch {release_branch_name}")
+
+
+def create_pull_request_for_upcoming_version_branch(main_branch, repository, upcoming_version,
+                                                    upcoming_version_branch):
+    print(f"### Creating pull request for {upcoming_version_branch}### ")
+    pr_result = repository.create_pull(title=f"Bump Citus to {upcoming_version}", base=main_branch,
+                                       head=upcoming_version_branch, body="")
+    print(f"### OK Pull request created. PR Number:{pr_result.number} PR URL: {pr_result.url}### ")
+
+
+def push_branch(upcoming_version_branch):
+    print(f"Pushing changes for {upcoming_version_branch} into remote origin...### ")
+    run(f"git push --set-upstream origin {upcoming_version_branch}")
+    print(f"### OK Changes pushed for {upcoming_version_branch}### ")
+
+
+def commit_all_changes_on_branch(project_name, project_version, release_branch):
+    print(f"### Committing changes for branch {release_branch}...### ")
+    run(f' git commit -a -m "Bump  {project_name} version to  {project_version} "')
+    print(f"### OK Changes committed for {release_branch}### ")
+
+
+def update_version_with_upcoming_version_in_citus_control(citus_control_file_path, upcoming_minor_version):
+    print(f"### Updating {citus_control_file_path} file with the upcoming version {upcoming_minor_version}...### ")
+    if not replace_line_in_file(citus_control_file_path, CITUS_CONTROL_SEARCH_PATTERN,
+                                f"default_version = '{upcoming_minor_version}-1'"):
+        raise ValueError(f"{citus_control_file_path} does not have match for version")
+    print(f"### OK{citus_control_file_path} file updated with the upcoming version {upcoming_minor_version}...### ")
+
+
+def update_schema_version_with_upcoming_version_in_multi_extension_file(current_schema_version,
+                                                                        multi_extension_sql_path,
+                                                                        upcoming_minor_version, upcoming_version):
+    print(
+        f"### Updating schema version {current_schema_version} on {multi_extension_sql_path} "
+        f"file with the upcoming version {upcoming_version}...### ")
+    # TODO Append instead of replace may require
```

Comment:
what does that mean ?

Reply:
CHangin schema version with the upcoming version param

Reply:
Yes, it's the case indeed.
We should add new lines with `ALTER EXTENSION citus UPDATE TO` commands, should not replace the previous ones.

Reply:
When I look these lines I see that there is a replace operation but when I look comment it was supposed to be append
I'm confused. Which one that should be?
https://github.com/citusdata/tools/blob/b55e5e717faf61ca3da4a00559c1ace6f469372c/automated_packaging/prepare_release.pl#L95-L96

Reply:
we need to append. I believe, above `sed` commands also do that

Reply:
Yes I've just realized that -i option appends line. I'll fix it

Reply:
then waiting for that item to be resolved, thanks

Reply:
Fixed it